skeleton/cli-bundler/aurelia_project/tasks/process-css.ext

```diff
@@ -76,6 +76,14 @@ export default function processCSS() {
       cssnano()
     ]))
     // @endif
+    // @if feat['postcss-tailwind']
+    .pipe(postcss([
+	  require('tailwindcss'),
```

Comment:
For consistence, can we move this commonjs require to an es6 import line at the top like other plugins? We are writing in es6 module, better to avoid mixing commonjs calls.

Reply:
The cli-bundler tailwind css setup didn't work for less/sass/style, because `project.cssProcessor.source` doesn't contain `*.css` files. the `tailwind.css` file is not processed.

Maybe have to use [build-time-imports](https://tailwindcss.com/docs/using-with-preprocessors/#build-time-imports) in less/scss/styl files?
```css
@import "tailwindcss/base";
```

Reply:
import lines fixed

Reply:
I'm not an expert on this, but I found it weird!
As before, everything works with Webpack.
A `tailwind.css` file for everthing and a `<require from="./tailwind.css"></require>` in `app.html`.
This makes sense.
As you mentioned, When the problem arises, the settings are
a) Cli-bundler + Less
b) Cli-bundler + Sass
c) Cli-bundler + Stylus
`project.cssProcessor.source` doesn't contain `*.css` files. It contains any of the following types
`.less`, `.scss` , `.styl`
For such circumstances I made the following changes:
a) 
```js
// @if feat['postcss-tailwind']
    .pipe(postcss([
      // @if feat['cli-bundler'] && (feat['less'] || feat['sass'] || feat['stylus'])
      postcssImport(),
      // @endif
      tailwindcss('tailwind.config.js'),
      autoprefixer(),
      postcssUrl({url: 'inline', encodeType: 'base64'}),
      cssnano()
    ]))
// @endif
```

b) Use one these 

`tailwind.less__if_postcss-tailwind_and_cli-bundler_and_less`
`tailwind.less`

`tailwind.scss__if_postcss-tailwind_and_cli-bundler_and_sass`
`tailwind.scss`

`tailwind.styl__if_postcss-tailwind_and_cli-bundler_and_stylus`
`tailwind.styl`

instead of 

`tailwind.css__if_postcss-tailwind_and_cli-bundler_and_not_less_and_not_sass_and_not_stylus`
`tailwind.css__if_postcss-tailwind_and_webpack`
`tailwind.css`

now `project.cssProcessor.source` has a file.

But what's weird here?

I **must** add 

```js

  <!-- @if feat['postcss-tailwind'] -->
  <require from="./tailwind.css"></require>
  <!--  @endif -->
```
`tailwind.css` even for `.less` or `.scss` or `.styl` in `app.html`.
Without this line, the final result of the bundler contains `tailwindcss` but does not render in the browser.
It looks like we need the above code at runtime.
Is this the right way? Doesn't it cause trouble somewhere? Unfortunately, I don't have much information on this.

Reply:
webpack didn't have issue because our webpack config has rules for `test: /\.css$/i,` and `test: /\.scss$/i,`.

I suspect in your option-b, less/sass compiler will raise error on the content of `tailwind.less` file.

Our cli-bundler default setup only process the scss file, skipped css file by default. We can enable both before calling postcss.

1. add css files to `project.cssProcessor.source` like this:
```
"cssProcessor": {
  "source": [
    "src/**/*.css",
    "src/**/*.scss"
  ]
}
```
2. conditional process it.
```js
import * as gulpIf form 'gulp-if';

.pipe(gulpIf(f => f.extname === '.scss', sass().on('error', sass.logError)))
```

Anyway, you can pick whatever approach works and looks clean.

Reply:
Yes, for cli-bundler always require `xxx.css` file even for `xxx.less` or `xxx.scss` because cli-bundler doesn't care or know the original file, it only sees the final compiled css file (sent over to cli-bundler by gulp pipeline). If this is your question.

Reply:
So the thing I said is not that weird! :D
I test the method you mentioned (`gulpIf`) today.
Thanks.

Reply:
It was a good question. Thanks for that, I realised this can be improved in dumber to be more intuitive for users from webpack. https://github.com/dumberjs/dumber-module-loader/commit/5d0aaffc7ab21ab497e0af1ff453b8be84b078ba

This is also possible with cli-bundler, but the architecture difference will make the implementation more difficult.

Reply:
I found your recent approach better and cleaner. If you have time please check the PR again. Thanks.